module/PowerNSX.psm1

```diff
@@ -36249,5 +36284,280 @@ function Copy-NsxEdge{
+
+    .EXAMPLE
+
+    Get-NsxEdge Edge01 | Get-NsxDns | Set-NsxDns -Enabled:$false
+
+    Disabled the DNS server on ESG
+
+    .EXAMPLE
+
+    Get-NsxEdge Edge01 | Get-NsxDns | Set-NsxDns -DNSServer 192.0.2.2
+
+    Set the DNS Server to 192.0.2.2
+
+    .EXAMPLE
+
+    Get-NsxEdge Edge01 | Get-NsxDns | Set-NsxDns -CacheSize 32
+
+    Change DNS Cache Size to 32 (Mb)
+
+    .EXAMPLE
+
+    Get-NsxEdge Edge01 | Get-NsxLoadBalancer | Set-NsxDNS -EnableLogging
+
+    Enabled DNS traffic logs.
+
+    .EXAMPLE
+
+    Get-NsxEdge Edge01 | Get-NsxLoadBalancer | Set-NsxDNS -LogLevel debug
+
+    Choose the log level (emergency, alert, critical, error, warning, notice, info, debug)
+    of DNS logs.
+
+    #>
+
+    param (
+        [Parameter (Mandatory=$true,ValueFromPipeline=$true,Position=1)]
+            [ValidateScript({ ValidateEdgeDNS $_ })]
+            [System.Xml.XmlElement]$DNS,
+        [Parameter (Mandatory=$False)]
+            [switch]$Enabled,
+        [Parameter (Mandatory=$False)]
+            [ValidateNotNullOrEmpty()]
+            [ipaddress[]]$DNSServer,
+        [Parameter (Mandatory=$False)]
+            [ValidateRange(1,8196)]
+            [int]$CacheSize,
+        [Parameter (Mandatory=$False)]
+            [ValidateNotNullOrEmpty()]
+            [switch]$EnableLogging,
+        [Parameter (Mandatory=$False)]
+            [ValidateSet("emergency","alert","critical","error","warning","notice","info","debug")]
+            [string]$LogLevel,
+        [Parameter (Mandatory=$False)]
+            #PowerNSX Connection object
+            [ValidateNotNullOrEmpty()]
+            [PSCustomObject]$Connection=$defaultNSXConnection
+    )
+
+    begin { }
+
+    process {
+        #Create private xml element
+        $_Dns = $DNS.CloneNode($true)
+
+        #Store the edgeId and remove it from the XML as we need to post it...
+        $edgeId = $_Dns.edgeId
+        $_Dns.RemoveChild( $((Invoke-XPathQuery -QueryMethod SelectSingleNode -Node $_Dns -Query 'descendant::edgeId')) ) | out-null
+
+        #Using PSBoundParamters.ContainsKey lets us know if the user called us with a given parameter.
+        #If the user did not specify a given parameter, we dont want to modify from the existing value.
+
+        if ( $PsBoundParameters.ContainsKey('Enabled') ) {
+            if ( invoke-xpathquery -node $_Dns -querymethod SelectSingleNode -Query "child::enabled" ) {
+                $_Dns.enabled = $Enabled.ToString().ToLower()
+            } else {
+                Add-XmlElement -xmlroot  $_Dns -xmlElementName "enabled" -xmlElementText $Enabled.ToString().ToLower()
+            }
+        }
+
+        if ( $PsBoundParameters.ContainsKey('CacheSize') ) {
+            if ( invoke-xpathquery -node $_Dns -querymethod SelectSingleNode -Query "child::cacheSize" ) {
+                $_Dns.CacheSize = $CacheSize
+            } else {
+                Add-XmlElement -xmlroot  $_Dns -xmlElementName "cacheSize" -xmlElementText $CacheSize
```

Comment:
Need to convert $CacheSize to a string using `$CacheSize.tostring()` otherwise this fails the tests on Powershell Windows.